### PR TITLE
Some tweaks to how packages are installed, plus removal of dhcpd5 dependency (Option 1)

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Some tweaks to the order in which how packages are installed, plus removal of dhcpd5 dependency. In the case that `dhcpcd5` is already installed (i.e on Rapsberry Pi OS), the user will be offered whiptail dialogs to configure it, the same as we currently offer.

Supercedes:
- https://github.com/pi-hole/pi-hole/pull/4258
- https://github.com/pi-hole/pi-hole/pull/4259
- https://github.com/pi-hole/pi-hole/pull/3715

Offered as an alternative to:
- https://github.com/pi-hole/pi-hole/pull/4261 (don't merge both!!)

### CODE REVIEW

File: advanced/Templates/pihole-FTL.service [hunk 1]
```diff
@@ -37,8 +37,9 @@ start() {
     chown pihole:pihole /etc/pihole /etc/pihole/dhcp.leases 2> /dev/null
     chown pihole:pihole /var/log/pihole-FTL.log /var/log/pihole.log
     chmod 0644 /var/log/pihole-FTL.log /run/pihole-FTL.pid /run/pihole-FTL.port /var/log/pihole.log
+    chmod 0644 /etc/pihole/macvendor.db
```
Comment (DL6ER) [inline advanced/Templates/pihole-FTL.service:40]: Is there a specific reason to change the owner/permission of this file? It is read-only for FTL and only updated on FTL updates (via the installer which is `root`)?

Comment (PromoFaux) [reply advanced/Templates/pihole-FTL.service:40]: We're already doing it, I just moved it from the `FTLInstall` function into the service file (and added the leading `0` to match the `chmod` above)

https://github.com/pi-hole/pi-hole/blob/master/automated%20install/basic-install.sh#L2325

The original intention was to move the call to `FTLdetect` higher in the install process, but that caused some different issues. However, in order to move it higher, we needed to get rid of it's usage of the `pihole` user.

**Comment by dschaper** [comment]
So #4258 is go/no-go for review?

**Comment by PromoFaux** [comment]
You can review if you like, but the changeset is also included here (and #4621)  as all these changes need to be made together in order to simplify things.

**Comment by dschaper** [comment]
Okay, if we do this in 5.4 then we need to make sure that the @pi-hole/debug team knows this is happening so they can adjust their approaches.

**Comment by DL6ER** [comment]
I (somehow) prefer this (configure `dhcpcd5` if already present) over the other (don't deal with `dhcpcd5` at all) because it makes the setup easier for the total beginners - those who just bought a Raspberry Pi for the first time.

**Comment by jfb-pihole** [comment]
> it makes the setup easier for the total beginners - those who just bought a Raspberry Pi for the first time.

I'm in agreement here.  Most (almost all) users are using a Pi for the first time to run Pi-hole.

**Comment by PromoFaux** [comment]
Appreciate comments, all. I'm generally in disagreement, but... I'm also in the minority camp so happy for it to go the way that most of us are comfortable with. Not worth me getting precious about :)

**Comment by dschaper** [comment]
Remember too, we can always go for Option 2 in the future if we find that users don't really need the hand holding. 

I really would have preferred to do this in two steps, move the `dhcpd` to installer dependency only and leave everything as is and then later do the change to support Raspbian only but I'll compromise to you as well and go with this that does it in one PR and one step.

**Comment by PromoFaux** [comment]
The biggest biggest biggest thing for me is to stop forcing `dhcpcd5` on users that are on non-Raspbian distros, which this covers, and it's something we've been vaguely promising we'd sort out for ages now, so I can sleep well at night knowing that we've struck a bugbear off the list 🎉

**Comment by yubiuser** [comment]
I support this too. It is a good compromise between forcing `dhcp5` to users who don't run raspbian and dropping it altogether. I'm not sure if the total support load will increase or not - so far we tell users to run `pihole -r` to reconfigure, now we need to tell them to set a static IP. Might be the same work load.

**Comment by DL6ER** [comment]
> Might be the same work load.

Given a good documentation about how to setup a static IP is readily available. (but I hear rumors something is being prepared)

**Review by DL6ER — APPROVED** [review]

**Comment by PromoFaux** [comment]
https://github.com/pi-hole/docs/pull/547 feel free to push changes/make comments

**Comment by MichaIng** [comment]
I just recognised, this has not been merged into the `development` branch, which should be done as well?

**Pull request merged**